packages/xsnap/src/object-inspect.js

```diff
@@ -0,0 +1,489 @@
+  return false;
+}
+
+function isWeakSet(x) {
+  if (!weakSetHas || !x || typeof x !== 'object') {
+    return false;
+  }
+  try {
+    weakSetHas.call(x, weakSetHas);
+    try {
+      weakMapHas.call(x, weakMapHas);
+    } catch (s) {
+      return true;
+    }
+    return x instanceof WeakSet; // core-js workaround, pre-v2.5.0
+  } catch (e) {}
+  return false;
+}
+
+function inspectString(str, opts) {
+  if (str.length > opts.maxStringLength) {
+    const remaining = str.length - opts.maxStringLength;
+    const trailer = `... ${remaining} more character${
+      remaining > 1 ? 's' : ''
+    }`;
+    return (
+      inspectString($slice.call(str, 0, opts.maxStringLength), opts) + trailer
+    );
+  }
+  // eslint-disable-next-line no-control-regex
+  const s = $replace.call(
+    $replace.call(str, /(['\\])/g, '\\$1'),
+    // eslint-disable-next-line no-control-regex
+    /[\x00-\x1f]/g,
+    lowbyte,
+  );
+  return wrapQuotes(s, 'single', opts);
+}
+
+function lowbyte(c) {
+  const n = c.charCodeAt(0);
+  const x = {
+    8: 'b',
+    9: 't',
+    10: 'n',
+    12: 'f',
+    13: 'r',
+  }[n];
+  if (x) {
+    return `\\${x}`;
+  }
+  return `\\x${n < 0x10 ? '0' : ''}${$toUpperCase.call(n.toString(16))}`;
+}
+
+function markBoxed(str) {
+  return `Object(${str})`;
+}
+
+function weakCollectionOf(type) {
+  return `${type} { ? }`;
+}
+
+function collectionOf(type, size, entries, indent) {
+  const joinedEntries = indent
+    ? indentedJoin(entries, indent)
+    : $join.call(entries, ', ');
+  return `${type} (${size}) {${joinedEntries}}`;
+}
+
+function singleLineValues(xs) {
+  for (let i = 0; i < xs.length; i += 1) {
+    if (indexOf(xs[i], '\n') >= 0) {
+      return false;
+    }
+  }
+  return true;
+}
+
+function getIndent(opts, depth) {
+  let baseIndent;
+  if (opts.indent === '\t') {
+    baseIndent = '\t';
+  } else if (typeof opts.indent === 'number' && opts.indent > 0) {
+    baseIndent = $join.call(Array(opts.indent + 1), ' ');
+  } else {
+    return null;
+  }
+  return {
+    base: baseIndent,
+    prev: $join.call(Array(depth + 1), baseIndent),
+  };
+}
+
+function indentedJoin(xs, indent) {
+  if (xs.length === 0) {
+    return '';
+  }
+  const lineJoiner = `\n${indent.prev}${indent.base}`;
+  return `${lineJoiner + $join.call(xs, `,${lineJoiner}`)}\n${indent.prev}`;
+}
+
+function arrObjKeys(obj, inspect) {
+  const isArr = isArray(obj);
+  const xs = [];
+  if (isArr) {
+    xs.length = obj.length;
+    for (let i = 0; i < obj.length; i += 1) {
+      xs[i] = has(obj, i) ? inspect(obj[i], obj) : '';
+    }
+  }
+  const syms = getOwnPropertySymbols(obj);
+  for (const key in obj) {
+    if (!has(obj, key)) {
+      // eslint-disable-next-line no-continue
+      continue;
+    }
+    if (isArr && String(Number(key)) === key && key < obj.length) {
+      // eslint-disable-next-line no-continue
+      continue;
+    }
+    if ($test.call(/[^\w$]/, key)) {
+      xs.push(`${inspect(key, obj)}: ${inspect(obj[key], obj)}`);
```

Comment:
Confused why `obj` is included in `${inspect(key, obj)}` but is not in the symbol version below: `${inspect(syms[j])}`

Reply:
The symbol cannot possibly be the same as the object, so no need to check it for circularity.

Reply:
But the symbol's value can refer to the object, can't it?

Reply:
> But the symbol's value can refer to the object, can't it?

As far as I read it, the `inspect(syms[j])` inspects the symbol itself, not the symbol property's value.